includes/routes/event/rss.php

```diff
@@ -0,0 +1,113 @@
+			$rss_feed .= $this->get_item( $event );
+		}
+		$rss_feed .= $this->get_rss_20_footer();
+
+		header( 'Content-Type: application/xml; charset=UTF-8' );
+		// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+		echo $rss_feed;
+		exit();
+	}
+
+	/**
+	 * Get the RSS 2.0 header.
+	 *
+	 * @return string
+	 */
+	private function get_rss_20_header( array $events ): string {
+		$header  = '<rss version="2.0" xmlns:atom="http://www.w3.org/2005/Atom" xmlns:ev="http://purl.org/rss/1.0/modules/event/">';
+		$header .= '	<channel>';
+		$header .= '		<title>' . esc_html__( 'WordPress.org Global Translation Events', 'gp-translation-events' ) . '</title>';
+		$header .= '		<link>' . esc_url( home_url( gp_url( '/events' ) ) ) . '</link>';
+		$header .= '		<description>' . esc_html__( 'WordPress.org Global Translation Events', 'gp-translation-events' ) . '</description>';
+		$header .= '		<language>en-us</language>';
+		$header .= '	 	<pubDate>' . esc_html( $this->document_pub_and_build_date( $events ) ) . '</pubDate>';
+		$header .= '		<lastBuildDate>' . esc_html( $this->document_pub_and_build_date( $events ) ) . '</lastBuildDate>';
+		$header .= '		<docs>https://www.rssboard.org/rss-specification</docs>';
+		$header .= '		<generator>' . esc_html__( 'Translation Events', 'gp-translation-events' ) . '</generator>';
+		$header .= '		<atom:link href="' . esc_url( home_url( gp_url( '/events/rss' ) ) ) . '" rel="self" type="application/rss+xml"/>';
+		return $header;
+	}
+
+	/**
+	 * Get the RSS 2.0 footer.
+	 *
+	 * @return string
+	 */
+	private function get_rss_20_footer(): string {
+		$footer  = '	</channel>';
+		$footer .= '</rss>';
+		return $footer;
+	}
+
+	private function get_item( Event $event ) {
+		$item  = '		<item>';
+		$item .= '			<title>' . esc_html( $event->title() ) . '</title>';
+		$item .= '			<link>' . esc_url( home_url( gp_url( gp_url_join( 'events', $event->slug() ) ) ) ) . '</link>';
+		$item .= '			<description>' . esc_html( $event->description() ) . '</description>';
+		$item .= '          <enclosure url="' . esc_url( Urls::event_image( $event->id() ) ) . '" type="image/png" length="1200" />';
+		$item .= '			<pubDate>' . esc_html( $event->updated_at()->format( DATE_RSS ) ) . '</pubDate>';
+		$item .= '			<ev:startdate>' . esc_html( $event->start()->format( DateTimeInterface::ATOM ) ) . '</ev:startdate>';
+		$item .= '			<ev:enddate>' . esc_html( $event->end()->format( DateTimeInterface::ATOM ) ) . '</ev:enddate>';
+		$item .= '			<guid>' . esc_url( home_url( gp_url( gp_url_join( 'events', $event->slug() ) ) ) ) . '</guid>';
+		$item .= '		</item>';
+		return $item;
+	}
+
+	/**
+	 * Get the most recent event's pub date.
+	 *
+	 * @param Event[] $events Array of events to use for the pub date.
+	 *
+	 * @return string|null
+	 */
+	private function document_pub_and_build_date( array $events ): ?string {
+		if ( empty( $events ) ) {
+			return null;
+		}
+
+		$pub_date = $events[0]->updated_at()->format( DATE_RSS );
```

Comment:
We could make `$pub_date` an instance of `DateTimeImmutable` instead of a string, which would then allow us to more easily compare it with `$event->updated_at()`, and we would not need to call `strtotime()`.

Reply:
Done [here](https://github.com/WordPress/wporg-gp-translation-events/pull/326/commits/4eccfd26a3a42bd3fc68b1b07ab45faee0721cf5). Thank you!